### ISSUE TITLE
RL reward integration: RewardFunction + TRL adapter (aligned with TruGEPA)

### ISSUE THREAD

**Issue by joshreini1** [truera/trulens#2628]
## Description

No RL integration. Trulens metrics (groundedness, relevance, safety, conversation quality) are natural reward signals for RLHF/RL fine-tuning, but there's no adapter to feed them into RL trainers. TRL's `PPOTrainer`/`GRPOTrainer` accept `reward_funcs` (callable → scalar reward) — a clean integration point; custom-reward-function wiring is a known pain point there.

## Proposed Change

Trulens already has this same architectural pattern for a different optimizer: `TruGEPA` wraps a trulens metric as a scalar **fitness** for evolutionary prompt optimization, handles the `(score, metadata)` return, and logs each evaluation. `RewardFunction` is the RL analog — metric → **reward** for policy gradient. This issue aligns with — and shares code with — `TruGEPA` rather than duplicating the pattern.

1. **Shared base**: factor the common "metric → scalar optimization signal + log" logic out of `TruGEPA` so both `TruGEPA` (fitness) and the new `RewardFunction` (reward) reuse it: handle the `(score, dict)` return format, batching, and trajectory logging in one place.
2. **`RewardFunction` interface** (new `src/apps/rl/` package mirroring `src/apps/gepa/`): `RewardFunction.from_metric(metric, *, aggregation=mean, transform=sigmoid_to_reward)` producing a callable `(prompts, completions, **kwargs) -> List[float]`. Handle the [0,1] → reward transform (e.g. `2*score - 1`), batching, and async.
3. **TRL adapter**: `TRLRewardAdapter` exposing trulens metrics as `reward_funcs` for `PPOTrainer` + `GRPOTrainer`. Per-completion reward mode + conversation-level reward mode (compose with conversation metrics for multi-turn RL).
4. **Logging alignment**: `TruGEPA` currently logs via `TruVirtual` + `Select.RecordCalls` (legacy). `RewardFunction` logs via `EVAL_ROOT` spans with a `reward` tag (new Metric/OTEL path). Migrate `TruGEPA` to the same `EVAL_ROOT`/OTEL logging for a consistent optimization-trajectory view (part of the feedback→metric migration).
5. **Example notebook**: `examples/expositional/frameworks/trl/trulens_reward.ipynb` — GRPO fine-tune using a trulens metric as reward. Realistic data, no hardcoded credentials.

## Usage Example

```python
from trulens.apps.rl import TRLRewardAdapter

reward_fn = TRLRewardAdapter.from_metric(f_groundedness, transform="2x-1")
trainer = GRPOTrainer(model=..., reward_funcs=reward_fn, ...)
trainer.train()
```

## Files to Reference
- `src/apps/gepa/trulens/apps/gepa/fitness.py` — `TruGEPA`, the existing metric→fitness adapter (factor shared base from here)
- `src/apps/gepa/` — package structure to mirror for new `src/apps/rl/`

## Acceptance Criteria
- A trulens metric can be passed as `reward_funcs` to `GRPOTrainer` and a small training loop runs; rewards are visible as `EVAL_ROOT` spans; `TruGEPA` and `RewardFunction` share the same metric-wrapping/logging base; `TruGEPA` logs via the new `EVAL_ROOT` path; notebook runs end-to-end.

## Dependencies
- The conversation-level reward mode depends on #2618, #2624, #2625 landing first. The single-turn mode is independent. Aligns with `TruGEPA` and the feedback→metric migration. `trl` would be an optional dependency (`@pytest.mark.optional`).

**Comment by maitraymukeshkumarmodi-aiml** [comment]
Hi @joshreini1, I would like to claim this issue to add the RL Reward Integration!        
                                                                                          
Implementation Plan:                                                                      
                                                                                          
1. Factor out the shared metric → scalar optimization signal + log base from              
src/apps/gepa/fitness.py to unify metric wrapping, batching, and trajectory logging.      
2. Build src/apps/rl/ mirroring src/apps/gepa/, creating RewardFunction and               
TRLRewardAdapter.from_metric(metric, transform=...) compatible with Hugging Face trl      
trainers (GRPOTrainer / PPOTrainer).                                                      
3. Wire logging to EVAL_ROOT spans with a reward tag for consistent OTEL optimization     
tracking.                                                                                 
4. Add examples/expositional/frameworks/trl/trulens_reward.ipynb and corresponding unit   
tests with @pytest.mark.optional.                                                         
                                                                                          
Claiming this issue and starting implementation!

**Comment by sfc-gh-jreini** [comment]
Hey @maitraymukeshkumarmodi-aiml - you are already assigned and PR #2652 is open. Will review.
